code/game/objects/effects/misc.dm

```diff
@@ -81,7 +81,7 @@
 
 /obj/effect/dummy/lighting_obj/Initialize(mapload, _color, _range, _power, _duration)
 	. = ..()
-	set_light(_range ? _range : light_range, _power ? _power : light_power, _color ? _color : light_color)
+	AddComponent(/datum/component/overlay_lighting, _range, _power, _color)
```

Comment:
Won't this emit actual light, as well? All atoms have a `light_power` of `1`.

If an atom has a `light_power` greater than `0` and a `light_range` greater than `0` it toggles on the light.

Reply:
Changed the system so that is no longer true.
Now there's a `light_system` variable which differentiates static from moving lights.
Instead of only manipulating component variables, the atoms hold their own (as in the old system) and the moving lights component listens to changes.
This should make it a bit more generic and expansible.